### PR TITLE
add new kube_endpoint_address metric

### PR DESCRIPTION
**Pull request by bavarianbidi**
**What this PR does / why we need it**:

As KSM shouldn't be responsible for computation of metrics, a new metric
is needed to expose available and unavailable addresses of an endpoint.

This new metric is generated as a future replacement for
kube_endpoint_address_available and kube_endpoint_address_not_ready.

Signed-off-by: Mario Constanti <mario@constanti.de>

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*

Increases the cardinality of KSM: sum of all IPs in an endpoint subset (`endpoints.subsets.addresses.ip` and `endpoints.subsets.notReadyAddresses.ip`)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Partially solves #1624

### CODE REVIEW

File: internal/store/endpoint.go [hunk 1]
```diff
@@ -151,6 +151,34 @@ func endpointMetricFamilies(allowAnnotationsList, allowLabelsList []string) []ge
...
+			"Information about Endpoint available and non available addresses.",
+			metric.Gauge,
+			"",
+			wrapEndpointFunc(func(e *v1.Endpoints) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, s := range e.Subsets {
+					for _, available := range s.Addresses {
+						ms = append(ms, &metric.Metric{
+							LabelValues: []string{available.IP},
+							LabelKeys:   []string{"available_ip"},
```
Comment (fpetkovski) [inline internal/store/endpoint.go:165]: Should we have a single `ready` label which has `true` or `false` values?

Comment (bavarianbidi) [reply internal/store/endpoint.go:165]: > Should we have a single ready label which has true or false values?

yep, make sense.
should we than change `available_ip` and `unavailable_ip` to `ip`?

Comment (fpetkovski) [reply internal/store/endpoint.go:165]: Yes, I think making that change would be consistent with the `ready` label.

**Pull request merged**